src/main/java/io/lighty/yang/validator/formats/JsTree.java

```diff
@@ -61,23 +60,23 @@ public void emitFormat() {
         for (final RevisionSourceIdentifier source : this.sources) {
             final Module module = this.schemaContext.findModule(source.getName(), source.getRevision())
                     .orElseThrow(() -> new NotFoundException("Module", source.getName()));
-            final List<Integer> removeChoiceQnames = new ArrayList<>();
-            SingletonListInitializer singletonListInitializer = new SingletonListInitializer(1);
+            final SingletonListInitializer singletonListInitializer = new SingletonListInitializer(1);
 
+            final SchemaInferenceStack schemaInferenceStack = SchemaInferenceStack.of(this.schemaContext);
             // Nodes
-            printLines(getChildNodesLines(singletonListInitializer, module, removeChoiceQnames));
+            printLines(getChildNodesLines(singletonListInitializer, module, schemaInferenceStack));
```

Comment:
Direct children of module - do we need stack?

Reply:
These methods here used recursive and go through all child nodes in schemaTree.

Reply:
OK